src/protocol.rs

```diff
@@ -37,7 +37,7 @@ pub const VERSION: i32 = 2;
 
 /// Available services. This enum describes all service types the nscd protocol
 /// knows about, though we only implement `GETPW*`, `GETGR*`, and `INITGROUPS`.
-#[derive(Debug, FromPrimitive)]
+#[derive(Clone, Copy, Debug, FromPrimitive, ToPrimitive)]
```

Comment:
Can we do some kind of static assert on the max value of `ToPrimitive` to make sure that we have a large enough set allocated in Config?

Reply:
I added something, thanks. I think I'd feel better if I checked all values of `RequestType` but I think this is good enough for now.